Client/mods/deathmatch/logic/CClientModel.cpp

```diff
@@ -70,6 +80,21 @@ bool CClientModel::Deallocate(void)
                 }
             }
         }
+        else if (m_eModelType == CCLIENTMODELOBJECT) {
+            CClientObjectManager* pObjectManager = g_pClientGame->GetManager()->GetObjectManager();
+            std::vector<CClientObject*>::const_iterator iter = pObjectManager->IterBegin();
+            for (; iter != pObjectManager->IterEnd(); iter++)
+            {
+                if ((*iter)->GetModel() == m_iModelID)
+                {
+                    if ((*iter)->IsStreamedIn())
+                    {
+                        (*iter)->StreamOutForABit();
+                    }
+                    (*iter)->SetModel(1337);
+                }
+            }
```

Comment:
Id suggest using `auto`, and all the new features.
```cpp
for (auto iter = pObjectManager->IterBegin(); iter != pObjectManager->IterEnd(); iter++)
{
    const CClientObject* pObject = *iter;
    // reset of the code here
}
```
A more cleaner solution would be to add a method to the object manager:
```cpp
const auto& GetObjects() { return m_objects; } // or whatever the member variable std::vector is. 
```
And just use a `range based for loop`:
```cpp
for (auto* pObject : pObjectManager->GetObjects())
{
    // code
}
```

Reply:
> E1087	no instance of function "SharedUtil::CMappedContainer<T, LIST_TYPE>::begin [with T=CClientObject *, LIST_TYPE=std::vector<CClientObject *, std::allocator<CClientObject *>>]" matches the argument list and object (the object has type qualifiers that prevent a match)

Reply:
Add these lines to the CMappedArray
```
typename LIST_TYPE::const_iterator begin() const { return m_List.begin(); }
typename LIST_TYPE::const_iterator end() const  { return m_List.end(); }
```

Next, you should make a getted const:
```
const auto& GetObjects() { return m_objects; }
```
And finally:
for (const auto pObject : pObjectManager->GetObjects())
{
    // code
}